os/drivers/lcd/mipi_lcd.c

```diff
@@ -349,5 +417,17 @@ FAR struct lcd_dev_s *mipi_lcdinitialize(FAR struct mipi_dsi_device *dsi, struct
 	}
 	priv->config->backlight(CONFIG_LCD_MAXPOWER);
 
+#if defined(CONFIG_LCD_SW_ROTATION)
+	uint8_t *mem = (uint8_t *)kmm_malloc((CONFIG_LCD_XRES * CONFIG_LCD_YRES * 2 + 1) * NUM_OF_LCD_BUFFER);	// each pixel is 8bit int
+	if (!mem) {
+		lcddbg("ERROR: LCD rotate buffer memory allocation failed\n");
+		free(mem);
```

Comment:
1. This statement is failed to allocate so that no memory to free. This line should be removed.
2. Even we fail to allocate the memory, it returns `&priv->dev`. Is it ok? No need to return NULL?